packages/evm/contracts/test/MockInputValidatorChecker.sol

```diff
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity >=0.8.27;
+
+import {
+    AdvancedChecker
+} from "@excubiae/contracts/checker/AdvancedChecker.sol";
+
+/// @title MockInputValidatorChecker.
+/// @notice Enclave Input Validator
+/// @dev Extends BaseChecker for input verification.
+contract MockInputValidatorChecker is AdvancedChecker {
+    /// @notice Validates input
+    /// @param subject Address to check.
+    /// @param evidence mock proof
+    /// @return True if proof is valid
+    function _checkMain(
```

Comment:
nit: do we want to add `_checkPre` and `_checkPost` mocks to reach a more complete code coverage?

Reply:
I am not sure we need this usecase yet as we don't quite know the shape of all the validations etc. we need to do. My feeling is that we can roll with this and refactor as we cement what an input validation looks like.

Reply:
Yeah I know but I needed the advanced policy to allow for multiple main checks.